packages/labs/react/src/create-component.ts

```diff
@@ -241,6 +243,22 @@ export function createComponent<
     tag = tagName;
   }
 
+  // Warn users when web components use reserved React properties
+  if (MODE === 'dev') {
+    for (const p in element.prototype) {
+      if (!(p in HTMLElement.prototype) && reservedReactProperties.has(p)) {
+        // Note, this effectively warns only for `ref` since the other
```

Comment:
Good point. I'd be ok just checking for `ref` then and adding a comment to `reservedReactProperties` that says if anything is added there that's _not_ on the HTMLElement.prototype, to add a check for it.

Reply:
I think the warning improvement is something we can do in a future PR :)

Reply:
With the condition as I see it now, `if (p in element.prototype && !(p in HTMLElement.prototype))`, I don't think this comment applies anymore.

The comment made sense in its former location where the check against reserved props happened nested after an `in HTMLElement.prototype` check.

Now, the loop is purely for looking up whether any reserved prop is present in the custom element's prototype, sans `HTMLElement`'s prototype, so we are actually going to find things besides `ref` if they are defined as prop on the custom element.

Reply:
I think practically, everything in `reservedReactProperties` besides `ref` exists on the `HTMLElement.prototype` is the observation, right?

Reply:
That is true. And in the original way the loop and condition was written, we used to loop through `p in elemenet.prototype` and filter out `!(p in HTMLElement.prototype)`, then look for reserved prop such that we would only catch `ref` to `console.warn`.

Now that we loop through the set of reserved props instead, we actually get the opportunity to find other things besides `ref` too. The code view that shows above this thread in the conversation view is out of date.

Reply:
Soooo are we good here? I feel like this change is appropriate so we don't have to manually add properties as they come

Reply:
I agree and like the loop through of the list. That makes the comment about only warning for `ref` out of date though so I think the comment can be removed.

Reply:
good point!

Reply:
Looking more closely, I think keeping the `!(HTMLElement.prototype)` in the check is a benefit.

Some properties like:
- localName
- style
- className
- children

Those will always be on any custom element. Theyre properties on `HTMLElement` as well as a react reserved properties.

Warnings on `HTMLElement` properties might be confusing when all custom elements extend  HTMLElement

Reply:
Seems fine for now. This is devmode, so we can always change this to a hard-coded list if that ends up being less fussy.